src/Raven.Server/Documents/Handlers/Processors/TimeSeries/TimeSeriesHandlerProcessorForGetTimeSeriesRanges.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Microsoft.Extensions.Primitives;
+using Raven.Client;
+using Raven.Client.Documents.Operations.TimeSeries;
+using Raven.Server.Documents.Includes;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Platform;
+
+namespace Raven.Server.Documents.Handlers.Processors.TimeSeries
+{
+    internal class TimeSeriesHandlerProcessorForGetTimeSeriesRanges : AbstractTimeSeriesHandlerProcessorForGetTimeSeriesRanges<DatabaseRequestHandler, DocumentsOperationContext>
+    {
+        public TimeSeriesHandlerProcessorForGetTimeSeriesRanges([NotNull] DatabaseRequestHandler requestHandler) : base(requestHandler)
+        {
+        }
+
+        protected override async ValueTask GetTimeSeriesRangesAndWriteAsync(DocumentsOperationContext context, string documentId, StringValues names, StringValues fromList, StringValues toList, int start, int pageSize,
+            bool includeDoc, bool includeTags, bool returnFullResults, CancellationToken token)
+        {
+            using (context.OpenReadTransaction())
+            {
+                (long Count, DateTime Start, DateTime End) stats;
+                foreach (var name in names)
+                {
+                    stats = context.DocumentDatabase.DocumentsStorage.TimeSeriesStorage.Stats.GetStats(context, documentId, name);
+                    if (stats == default)
+                    {
+                        HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
+                        return;
+                    }
+                }
+
+                var includesCommand = includeDoc || includeTags
+                    ? new IncludeDocumentsDuringTimeSeriesLoadingCommand(context, documentId, includeDoc, includeTags)
+                    : null;
+
+                var ranges = GetTimeSeriesRangeResults(context, documentId, names, fromList, toList, start, pageSize, includesCommand, returnFullResults);
+
+                var actualEtag = CombineHashesFromMultipleRanges(ranges);
```

Comment:
do we take into account the includes?

Reply:
probably we didn't take them into account in the original code as well. 
Please check this and if this doesn't work as expected please open an issue.

Reply:
No they're currently not calculated in the hash
opened issue: https://issues.hibernatingrhinos.com/issue/RavenDB-18595